### PR TITLE
Externalised credentials

### PR DESCRIPTION
**Pull request by marcocast**

### CODE REVIEW

**Comment by marcocast** [comment]
please run the test again

**Comment by marcocast** [comment]
I said run the test again

**Comment by marcocast** [comment]
run the tests please

**Comment by marcocast** [comment]
please...

**Comment by marcocast** [comment]
ok what about I commit something random and you run the test again? how does it sound?

**Comment by marcocast** [comment]
retest this please

**Comment by marcocast** [comment]
ahah i knew it

**Comment by paraita** [comment]
Code is good ! although we can use some coding rules to make it [pep8 compliant](https://www.python.org/dev/peps/pep-0008/).

**Review by paraita — APPROVED** [review]

**Comment by paraita** [comment]
You can push to prod directly

**Comment by marcocast** [comment]
@paraita do you know any tools to make it pep8 compliant ?

**Comment by paraita** [comment]
from memory we used pylint in the past in London

**Comment by marcocast** [comment]
retest this please

**Comment by marcocast** [comment]
retest this please

File: proactive/ProActiveGateway.py [hunk 1]
```diff
@@ -1,39 +1,64 @@
...
+    def getJobState(self,job_id):
+        return self.proactive_schduler_client.getJobState(job_id).getName();
     
+    def isJobFinished(self,job_id):
+        return self.proactive_schduler_client.isJobFinished(job_id);
+    
+    def getJobInfo(self,job_id):
+        return self.proactive_schduler_client.getJobInfo(str(job_id));
+    
+    def getAllJobs(self, max_number_of_jobs=1000):
```
Comment (fviale) [inline proactive/ProActiveGateway.py:59]: if you mean to return all running jobs, you should name it getRunningJobs

**Review by fviale — APPROVED** [review]
Ok for this pull request, but I think you should implement the waitJob methods. To make things a bit more coherent, it would be best to have a real use case.

File: proactive/ProActiveGateway.py [hunk 2]
```diff
@@ -1,39 +1,64 @@
...
+        self.proactive_schduler_client = self.gg.jvm.org.ow2.proactive_grid_cloud_portal.smartproxy.RestSmartProxyImpl()
+        connection_info = self.gg.jvm.org.ow2.proactive.authentication.ConnectionInfo(self.base_url + "/rest", username, password, credentials_file, insecure)
         self.proactive_schduler_client.init(connection_info);
-    
-    def submitFromCatalog(self, bucket_name, workflow_name):
         
-        return self.proactive_schduler_client.submitFromCatalog(self.base_url + "/catalog",bucket_name, workflow_name).getReadableName();
+    def disconnect(self):
+        self.proactive_schduler_client.disconnect()        
+
```
Comment (fviale) [inline proactive/ProActiveGateway.py:37]: I understand that you restrict the scope somehow to the job level
in the implementations, we lack some functionalities:
- wait methods: like waitForJob, waitForAllJobs, waitForAnyJob.
- getting results: getTaskResult, getJobResult (here we can't skip the task level)
- kill, pause, resume, remove Job

File: proactive/ProActiveGateway.py (hunk 1 above)
Comment (marcocast) [reply proactive/ProActiveGateway.py:59]: no, I want to return ALL the jobs whatever is the status. It is then up to the user to filter out what he needs

File: proactive/ProActiveGateway.py (hunk 2 above)
Comment (marcocast) [reply proactive/ProActiveGateway.py:37]: Yes, we can create other tasks to increase other functionalities

File: proactive/ProActiveGateway.py (hunk 1 above)
Comment (fviale) [reply proactive/ProActiveGateway.py:59]: ok, but then if you want to return all jobs, you should not do JobFilterCriteria(False, False, True, False), but  JobFilterCriteria(False, True, True, True)

**Pull request merged**